### PR TITLE
Enclose all exposed classes in namespaces

### LINKED ISSUE #130: Namespace for Tests
Tests should also be placed in C++ namespaces.

This way we can keep them included in doxygen builds but they do not pollute the root scope.

Please see http://www.openpmd.org/openPMD-api/annotated.html for reference :)

### PR DESCRIPTION
**Pull request by anokfireball**
Put the classes inside test cases into namespaces as they are exposed
to the public.

Closes #130.

### CODE REVIEW

File: test/AuxiliaryTest.cpp [hunk 1]
```diff
@@ -54,20 +54,31 @@ TEST_CASE( "string_test", "[auxiliary]" )
     REQUIRE(expected3 == split("/path/to/relevant/data/", "/"));
 }
 
+namespace openPMD
+{
+namespace Test
```
Comment (ax3l) [inline test/AuxiliaryTest.cpp]: let us rename `Test` -> `test` for sanity :)

File: test/AuxiliaryTest.cpp [hunk 2]
```diff
@@ -54,20 +54,31 @@ TEST_CASE( "string_test", "[auxiliary]" )
...
+{
+struct S : public Writable
```
Comment (ax3l) [inline test/AuxiliaryTest.cpp]: maybe intend by one level here.

File: test/AuxiliaryTest.cpp [hunk 3]
```diff
@@ -54,20 +54,31 @@ TEST_CASE( "string_test", "[auxiliary]" )
...
+    Container< openPMD::Test::S > c = Container< openPMD::Test::S >();
     c.IOHandler = AbstractIOHandler::createIOHandler("", AccessType::CREATE, Format::DUMMY);
 
     REQUIRE(c.size() == 0);
     REQUIRE(c.erase("nonExistentKey") == false);
 }
 
+namespace openPMD
+{
+namespace Test
```
Comment (ax3l) [inline test/AuxiliaryTest.cpp]: `test`

**Comment by ax3l** [comment]
@C0nsultant can you please apply the suggested changes? :)

File: test/AuxiliaryTest.cpp (hunk 2 above)
Comment (anokfireball) [reply test/AuxiliaryTest.cpp]: Indenting the ```struct``` would make this one class inconsistent with all the rest of the API. I'm for leaving this one unchanged.

Comment (ax3l) [reply test/AuxiliaryTest.cpp]: ok I see, then we can make such changes later on - at once and consistently.
Here is how we usually style nested namespaces: https://github.com/ComputationalRadiationPhysics/contributing/blob/master/codingGuideLines/cpp.md#6-namespaces

A namespace only causes intend on nested non-namespaces :)

Comment (anokfireball) [reply test/AuxiliaryTest.cpp]: I do not dislike the style, but that will require a change to every single C++ source file in this project.

File: test/AuxiliaryTest.cpp [hunk 4]
```diff
@@ -78,9 +89,12 @@ class structure : public Attributable
     std::string text() const { return variantSrc::get< std::string >(getAttribute("text").getResource()); }
     structure& setText(std::string text) { setAttribute("text", text); return *this; }
 };
+} // Test
```
Comment (ax3l) [inline test/AuxiliaryTest.cpp]: // test

File: test/AuxiliaryTest.cpp [hunk 5]
```diff
@@ -54,20 +54,31 @@ TEST_CASE( "string_test", "[auxiliary]" )
...
+namespace openPMD
+{
+namespace test
+{
+struct S : public Writable
+{
+    int val;
+    bool written;
+};
+} // Test
```
Comment (ax3l) [inline test/AuxiliaryTest.cpp]: // test

File: test/AuxiliaryTest.cpp [hunk 6]
```diff
@@ -139,9 +157,12 @@ struct Widget : public Writable
     Widget(int)
     { }
 };
+} // Test
```
Comment (ax3l) [inline test/AuxiliaryTest.cpp]: // test

File: test/AuxiliaryTest.cpp [hunk 7]
```diff
@@ -180,9 +205,12 @@ class AttributedWidget : public Attributable
         return getAttribute(key).getResource();
     }
 };
+} // Test
```
Comment (ax3l) [inline test/AuxiliaryTest.cpp]: // test

File: test/AuxiliaryTest.cpp [hunk 8]
```diff
@@ -227,10 +259,12 @@ class Dotty : public Attributable
     Dotty& setAtt2(double d) { setAttribute("att2", d); return *this; }
     Dotty& setAtt3(std::string s) { setAttribute("att3", s); return *this; }
 };
+} // Test
```
Comment (ax3l) [inline test/AuxiliaryTest.cpp]: // test

**Review by ax3l — APPROVED** [review]

**Pull request merged**